code/mob/living/silicon/ai/holograms.dm

```diff
@@ -1,26 +1,96 @@
 //AI HOLOGRAMS
-/mob/living/silicon/ai/proc/create_hologram()
-	if (!deployed_to_eyecam)
-		boutput(src, "Deploy to an AI Eye first to create a hologram.")
-		return
-	if (length(holograms)>= max_holograms)
-		boutput(eyecam, "Not enough RAM to project more holograms. Delete others to make room.")
-		return
-	//select from list of icon_state in holograms.dmi
-	var/input = input("Select a type of hologram to display.","Hologram") as null|anything in list("caution","o2","beepsky","up_arrow","down_arrow","left_arrow","right_arrow","happy_face","neutral_face","sad_face","angry_face")
-	if (input)
+/mob/living/silicon/ai
+	contextLayout = new /datum/contextLayout/experimentalcircle(36)
+
+	proc/create_hologram()
+		if (!deployed_to_eyecam)
+			boutput(src, "Deploy to an AI Eye first to create a hologram.")
+			return
+		if (length(holograms)>= max_holograms)
+			boutput(eyecam, "Not enough RAM to project more holograms. Delete others to make room.")
+			return
+		//select from list of icon_state in holograms.dmi
+		var/input = input("Select a type of hologram to display.","Hologram") as null|anything in hologramTypes
```

Comment:
do we still want the input() way in? why not just have it via the point/context way?

Reply:
Made the button place the context menu on the current ai eye's position instead, removing the input and hologramTypes